workflow/table_generation.py

```diff
@@ -0,0 +1,800 @@
+    styleContents = styleContents.replace("$$$_HEATMAP_RANGE_$$$", heatmapCols)
+    # $$$_REV_ACTIVITY_COLS_$$$
+    revActivityCols = ",".join(
+        reversed(["ActivitySummary%s" % time for time in times]))
+    styleContents = styleContents.replace(
+        "$$$_REV_ACTIVITY_COLS_$$$", revActivityCols)
+    # $$$_BACKGROUND_COLS_$$$
+    bgCols = ",".join(["HeatMapBg%d" % (i+1) for i in range(timepoints-1)])
+    styleContents = styleContents.replace("$$$_BACKGROUND_COLS_$$$", bgCols)
+
+    with open(outStyleFile, "w") as f:
+        f.write(styleContents)
+
+    return pepsPerProt["all"]
+
+
+def RobustLog2(val, default):
+    """If the value is 0, return the default.  Otherwise return log2(val)."""
+    if val == 0:
+        return default
+    else:
+        return np.log2(val)
+
+# Copied from generate_prizes.py in TPS codebase
+# Sets a cap on the maximum prize by mapping p-values below 1E-10 to 1E-10
+# This allows for p-values that are not real p-values but rather fold change
+# threhsolds that were mapped to 0 or 1 "p-values"
+
+
+def LoadScores(firstfile, prevfile):
+    """Load the first and previous scores.  For each peptide, compute a prize
+    that is -log10(min p-value across all time points).  Assumes the scores
+    are p-values or equivalaent scores in (0, 1].  p-values < 1E-10 are mapped
+    to 1E-10.  Do not allow null or missing scores.
+
+    Return: data frame with scores and prize for each peptide
+    """
+    first_df = pd.read_csv(firstfile, sep="\t",
+                           comment="#", header=None, index_col=0)
+    prev_df = pd.read_csv(prevfile, sep="\t", comment="#",
+                          header=None, index_col=0)
+    first_shape = first_df.shape
+    assert first_shape == prev_df.shape, "First and previous score files must have the same number of peptides and time points"
+
+    assert not first_df.isnull().values.any(
+    ), "First scores file contains N/A values.  Replace with 1.0"
+    assert not prev_df.isnull().values.any(
+    ), "Previous scores file contains N/A values.  Replace with 1.0"
+
+    print("Loaded {} peptides and {} scores in the first and previous score files".format(
+        first_shape[0], first_shape[1]))
+
+    # Merge the two types of scores
+    merged_df = pd.concat([first_df, prev_df], axis=1, join="outer")
+    merged_shape = merged_df.shape
+    assert merged_shape[0] == first_shape[0], "First and previous significance scores contain different peptides"
+    assert merged_shape[1] == 2 * \
+        first_shape[1], "Unexpected number of significance scores after merging first and previous scores"
+
+    # Compute prizes
+    merged_df["prize"] = merged_df.apply(CalcPrize, axis=1)
+    return merged_df
+
+
+def CalcPrize(row):
+    """Compute the peptide prize as -log10(min p-value).  If min p-value
+    < 1E-10, set it to 1E-10."""
+    return -np.log10(max(min(row), 1E-10))
+
+
+def PrepTemporalCytoscape(peptideMapFile, timeSeriesFile, peptidePrizeFile,
```

Comment:
There are two versions of this function.  Do we need both?

Reply:
We reviewed this `PrepTemporalCytoscape` function and decided that we can delete it.  It was a simpler function that hard-coded more information and used pre-generated files instead of the TPS input files.  The above copy `PrepTemporalCytoscapeTPS` is more general.